cmd/pulsewatcher/pulsewatcher.go

```diff
@@ -156,57 +251,16 @@ func main() {
 		}
 		wg.Wait()
 
-		table := tablewriter.NewWriter(buffer)
-		table.SetHeader([]string{
-			"URL",
-			"Network State",
-			"Node State",
-			"Pulse Number",
-			"Active List Size",
-			"Working List Size",
-			"Role",
-			"Error",
-		})
-		table.SetBorder(false)
-
-		table.ClearRows()
-		table.ClearFooter()
-
-		moveBack(buffer)
-		buffer.Reset()
-
-		stateString := insolarReady
-		color := tablewriter.FgHiGreenColor
-		if !state || errored == len(conf.Nodes) {
-			stateString = insolarNotReady
-			color = tablewriter.FgHiRedColor
+		notReady := !state || errored == len(conf.Nodes)
+		if useJsonFormat {
+			displayResultsJson(results, buffer, notReady)
+		} else {
+			displayResultsTable(results, buffer, notReady)
 		}
 
-		table.SetFooter([]string{
-			"", "", "", "",
-			"Insolar State", stateString,
-			"Time", time.Now().Format(time.RFC3339),
-		})
-		table.SetFooterColor(
-			tablewriter.Colors{},
-			tablewriter.Colors{},
-			tablewriter.Colors{},
-			tablewriter.Colors{},
-
-			tablewriter.Colors{},
-			tablewriter.Colors{color},
-
-			tablewriter.Colors{},
-			tablewriter.Colors{},
-		)
-
-		lock.Lock()
-		table.AppendBulk(results)
-		lock.Unlock()
-
-		table.Render()
-
-		fmt.Print(buffer)
+		if singleOutput {
+			break
```

Comment:
can we move a large piece of code that starts in 'for' in another function, that collects data?

Reply:
done.